rtems/Mutex.java

```diff
@@ -49,25 +58,110 @@ public void lock() throws InterruptedException{
 				//how should we prepend here???Doubt the orderRec as it is already present in thisThread.mutexList
 			}
 		  }finally{
-		  	parentLock.unlock();
+			parentLock.unlock();
 		  }
 
 	}
 	public void unlock() throws InterruptedException{
+		Object topMutex=null;
```

Comment:
I have made changes as per your suggestion. Please have a look at it in updated code.